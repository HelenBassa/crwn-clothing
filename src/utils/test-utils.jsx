import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };