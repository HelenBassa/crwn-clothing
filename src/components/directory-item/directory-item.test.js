import { render, screen } from "../../utils/test-utils";
import DirectoryItem from "./directory-item.component";

test("should render DirectoryItem component", () => {
  const mockCategory = {
    title: "hats",
    imageUrl: "https://i.ibb.co/cvpntL1/hats.png",
    route: "shop/hats",
  };

  const { container } = render(<DirectoryItem category={mockCategory} />);
  expect(container).toMatchSnapshot();

  expect(screen.getByText("hats")).toBeInTheDocument();
  expect(screen.getByText(/shop now/i)).toBeInTheDocument();
});
