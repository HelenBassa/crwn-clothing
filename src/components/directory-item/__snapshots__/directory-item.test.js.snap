// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DirectoryItem component 1`] = `
<div>
  <div
    class="sc-dkrFOg kupPSc"
  >
    <div
      class="sc-bcXHqe mwYYS"
    />
    <div
      class="sc-gswNZR cGctEJ"
    >
      <h2>
        hats
      </h2>
      <p>
        Shop now
      </p>
    </div>
  </div>
</div>
`;
